docs/src/docs/common-pitfalls.md

```diff
@@ -0,0 +1,187 @@
+
+There are two situations where you may encounter this pitfall:
+
+- Permalinked static files that aren't HTML (`sitemap.xml`, `robots.txt`, and the like).
+- Resources that are passthrough-copied (fonts, images, etc.).
+
+Let's take a closer look at each scenario.
+
+### Permalinked static files that aren't `.html`
+
+Suppose you're using 11ty to auto-generate a `sitemap.xml` for your site. In 11ty, you'd create this file using your chosen templating language extension and set a root-relative permalink in its front matter, like so:
+
+```liquid
+<!-- src/sitemap.liquid -->
+---
+permalink: /sitemap.xml
+---
+```
+
+When _plain_ 11ty builds your site, it processes this template file and spits out a file named `_site/sitemap.xml`.
+
+⚠️ **However, this won't work in Slinkity production builds!** [Since we build to a temporary output](/docs/how-we-build-your-site), the sitemap will get written to `.11ty-build-<hash>/sitemap.xml`. In the follow-up steps, Vite will:
+
+1. Process this temporary build directory.
+2. See that a) `sitemap.xml` is not referenced by any other file and b) isn't in the dedicated `public` folder.
+3. Exclude it from the final output that it writes to `_site`.
+
+#### Solution
+
+To fix this problem, you need to prefix the static file's permalink with `/public`, like this:
+
+```liquid
+<!-- src/sitemap.liquid -->
+---
+permalink: /public/sitemap.xml
+---
+```
+
+Now, this happens:
+
+1. 11ty processes the template and writes it to a `public` folder nested inside the temporary build folder: `.11ty-build-<hash>/public/sitemap.xml`.
+2. Vite sees the `public` folder and copies it into your final build output directory, giving you `_site/sitemap.xml`. Note that the nested `/public` directory disappears from the final build output!
+
+### Resources that are passthrough-copied
+
+A typical 11ty site has many static resources, including fonts, images, and more. Some of these (like images) may need to be processed manually using 11ty, but others (like fonts) just need to be copied to the build output directory.
+
+In Slinkity, there are two approaches you can take to ensure that static resources correctly make their way to the final build output:
+
+1. Using relative paths or import aliases to the resources on a page.
+2. Passthrough-copying the resources to a `/public` folder.
+
+#### Approach 1: Path resolution
+
+In the first approach, we lean on Vite's ability to resolve dependencies and resources referenced by our templates. In this case, if you want your font files to end up in your site's final build output, all you need to do is reference those font files somewhere in your CSS. You can do this using either relative paths (not absolute!) or [import aliases](/docs/import-aliases).
+
+For example, suppose you have a project structure that looks like this:
+
+```plaintext
+├── fonts
+│   └── Atkinson-Hyperlegible-Regular.woff2
+├── src
+├── _includes
+│   │   └── layout.njk
+│   ├── index.md
+├── styles
+│   ├── index.css
+```
```

Comment:
On my site, so that I don't have to remember to add `plaintext` for plaintext code blocks, I opted to remove 11ty's syntax highlighter plugin and replace it with markdown-it-prism, setting the default language to `plaintext`:

https://github.com/AleksandrHovhannisyan/aleksandrhovhannisyan.com/blob/master/config/plugins/markdown.js#L15

Thoughts on doing this?